### DISCUSSION TITLE
Simulating Tabs on Android

### DISCUSSION THREAD

**Discussion by canedha** [beeware/toga#1257] [Q&A]
Hi,

as Toga has no Optioncontainers available for android, I thought I could simulate this behaviour with creating different boxes withwin an main box/window and simply hide/show the "option" selected.

When I try to run my code, I get the error on buttonpress "Already have a container". How can I hide a box and show another in the same window?

Here's my code so far:

`import toga
from toga.style import Pack
from toga.fonts import Font, SANS_SERIF, SERIF
from toga.style.pack import COLUMN, ROW, CENTER

class HelloWorld(toga.App):
    def startup(self):

        self.box1 = toga.Box(style=Pack(direction=COLUMN))
        
        button1 = toga.Button(
            'Switch to 2!',
            on_press=self.switch2,
            style=Pack(padding=(5,80))
        )
        self.box1.add(button1)
        
        self.box2 = toga.Box(style=Pack(direction=COLUMN))
        
        button2 = toga.Button(
            'Switch to 1!',
            on_press=self.switch1,
            style=Pack(padding=(5,80))
        )
        self.box2.add(button2)
        
        
        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = self.box1
        self.main_window.show()

    def switch2(self, widget):
        self.main_window.remove = self.box1
        self.main_window.content = self.box2
        self.main_window.show()
        
    def switch1(self, widget):
        self.main_window.remove = self.box2
        self.main_window.content = self.box1
        self.main_window.show()

def main():
    return HelloWorld()`

Any suggestions?

**Comment by eichin** [comment]
FYI `OptionContainer` works on Android now - but I think this example would work if in the `switch()` functions you instead used `self.main_window.add(self.box1)` &c. because https://toga.readthedocs.io/en/stable/reference/api/widgets/widget.html#toga.Widget documents `add()` as doing the rest of the work:

> If a child widget already has a parent, it will be re-parented as a child of this widget. If the child widget is already a child of this widget, there is no change.
